header_guard.py

```diff
@@ -0,0 +1,108 @@
+from __future__ import annotations
+from pathlib import Path
+import argparse
+import re
+from typing import Optional, Tuple
+
+HEADER_EXTS: set[str] = {".h", ".hpp", ".hh", ".hxx", ".inl", ".tpp"}
+LEAD_RE = re.compile(r'^(?:\s*//[^\n]*\n|/\*.*?\*/\s*|\s*\n)+', re.DOTALL)
```

Comment:
RE for regular expression is hard to understand. Please use RegEx or RegularExpression instead for all variables in this implementation.